### PR TITLE
Ft requests

### PR DESCRIPTION
**Pull request by RachaelNantale**

### CODE REVIEW

File: app/utility.py [hunk 1]
```diff
@@ -1,55 +1,66 @@
...
 
+    if time is None or len(time) < 3 or len(time) > 10:
+        errors.append(
+            'Time input must be provided and should be between 3 and 10 \
+                characters long and dont use symbols')
+        result = False
 
-class ValidateUserData:
-    """ Class to validate user authentication"""
+    if time.alnum():
```
Comment (isheebo) [inline app/utility.py:42]: use a different name instead of time. `time` is a python module

File: app/utility.py [hunk 2]
```diff
@@ -1,55 +1,66 @@
...
-        if not bool(re.fullmatch('^[A-Za-z ]*$', name)):
-            return {"status": False,
-                    "message": "Invalid characters not allowed"}
+def validate_user_input(self, username, email, password, phone, errors):
+    result = True
 
-        if not re.match(r"([\w\.-]+)@([\w\.-]+)(\.[\w\.]+$)", email):
-            return {"status": False, "message": "Enter valid email "}
+    if re.compile('[!@#$%^&*:;?><.0-9]').match(username):
+        errors.append('Invalid characters not allowed')
```
Comment (isheebo) [inline app/utility.py:51]: make this message more descriptive. saying invalid characters doesn't tell the user which characters are actually invalid

File: app/api/models/request.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        self.time = time
+        self.status = status
+        self.errors = []
+
+    def to_json(self):
+        """
+        json representation of the Ride model
+        """
+        return {
+            'Request id': self.request_id,
```
Comment (isheebo) [inline app/api/models/request.py:30]: instead of saying `Request id`, saying `request_id` would make your work much easier

Comment (isheebo) [reply app/api/models/request.py:30]: The same idea applies for all json keys

File: app/api/models/request.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            'Ride id': self.ride_id,
+            'Passenger': self.passenger,
+            'Pickup Point': self.pickup_point,
+            'Destination': self.destination,
+            'Time': self.time,
+            'Status': self.status
+        }
+
+    def save_to_db(self):
+        # if validate_ride_input(self,  self.passenger, self.pickup_point,
```
Comment (isheebo) [inline app/api/models/request.py:40]: remove unused/commented code

File: app/utility.py (hunk 1 above)
Comment (geofrocker) [reply app/utility.py:42]: Are using this file?

Comment (RachaelNantale) [reply app/utility.py:42]: @geofrocker  no i removed it

**Pull request merged**